dmoj/judge.py

```diff
@@ -39,6 +37,7 @@ class IPC(Enum):
     RESULT = 'RESULT'
     BATCH_BEGIN = 'BATCH-BEGIN'
     BATCH_END = 'BATCH-END'
+    PRETEST_END = 'PRETEST-END'
```

Comment:
We should also add a `PRETEST-BEGIN` packet so that the state machine on the bridge side is trivial to write.

Reply:
Ok, only question is, if there are no pretests, are these packets sent?

Reply:
Ideally they shouldn't be, but I don't think it'd change bridge-side logic much if they were. Just a small efficiency thing to not send them at all if we don't have pretests.